qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -219,6 +165,11 @@ def show(self):  # noqa
         self.window.show()
         self.window.activateWindow()
         self.window.raise_()
+        if self.window.windowState() & Qt.WindowMinimized:
```

Comment:
Ah yes, this is a bit odd. On Windows a window can be both minimised and maximised at the same time (i.e. clicking it in the taskbar will restore it maximised). A combination of states can be set hence the bitwise check to see if it is minimised. Then if it is also maximised it gets restored maximised, else it gets restored normally. I'll add a comment explaining this. Perhaps it is possible to avoid the maximised check, but I think because `setWindowState()` can not take `Qt.WindowActive` as an argument this would get messy. http://doc.qt.io/qt-5/qt.html#WindowState-enum

Reply:
I see. That still feels odd but it makes sense once you explain it. As you say, a comment is probably a good idea there.